apps/trip_plan/lib/trip_plan/api/open_trip_planner/parser.ex

```diff
@@ -77,10 +77,14 @@ defmodule TripPlan.Api.OpenTripPlanner.Parser do
   end
 
   defp parse_time(ms_after_epoch) do
-    ms_after_epoch
-    |> DateTime.from_unix!(:millisecond)
-    |> Timex.set(microsecond: {0, 0})
-    |> Timex.to_datetime(OTP.config(:timezone))
+    case ms_after_epoch
+         |> DateTime.from_unix!(:millisecond)
+         |> Timex.set(microsecond: {0, 0})
+         |> DateTime.to_unix()
+         |> FastLocalDatetime.unix_to_datetime(OTP.config(:timezone)) do
+      {:ok, ms_after_epoch_dt} -> ms_after_epoch_dt
+      _ -> :error
```

Comment:
This should only fail if the timezone is invalid, so it's okay to do a raw match (which will crash if the timezone is invalid):
```
{:ok, dt} = ms_after_epoch |> ...
dt
```